```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Setting libraries

library("phyloseq")
library("ggplot2")
library("scales")
library("grid")
library("vegan")
library("ape")
library("plyr")
library("ggpmisc")
library("dplyr")
library("broom")
library("picante")
library("Rmisc")
library("emmeans")
library("car")
library("lme4")
library("tibble")
library("data.table")
library("microbiome")
library("ggvenn")
library("patchwork")
library("RColorBrewer")
library("ampvis2")
library("microViz")
library("tidyverse")
library("tidyr")
library("parallel")
library("tidyr")
library("stringr")
library("stringi")
library("MuMIn")
library("ggpubr")
#library("NetCoMi")
library("randomcoloR")
library("miaViz")
library("cowplot")
library("decontam")
library(ggthemes)
library(ggVennDiagram)
library(janitor)

```
# ANALYSIS FOR ALL FACILITIES
Final analysis with all facility types and POnd1-Season2 from FacilityB
modified from Cantoran et al. 2023 (DOI: 10.1093/femsec/fiad098)

```{r, warning=FALSE, message=FALSE, echo=FALSE}

### Read phyloseq pbject
ps.16S.nc <- readRDS('ps.16S.new.rds')
ps.ITS.nc <- readRDS('ps.ITS.new.rds')

```

# Roots
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Sub-setting season 1 and 1 Pond from GCG
ps1.16S.R <- ps.16S.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "Roots")

ps1.ITS.R <- ps.ITS.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "Roots")

# CSS Normalization
#Convert the phyloseq object to a metagenomeseq object:
mgs_cssBR <- phyloseq_to_metagenomeSeq(ps1.16S.R)
mgs_cssFR <- phyloseq_to_metagenomeSeq(ps1.ITS.R)

# Perform the Cumulative Sum Scaling:
mgs_cssBR <- cumNorm(mgs_cssBR)
mgs_cssFR <- cumNorm(mgs_cssFR)

# Extract the counts and add them to a separate phyloseq object:
css_countsBR <- MRcounts(mgs_cssBR, norm = TRUE)
css_countsFR <- MRcounts(mgs_cssFR, norm = TRUE)

ps_cssBR <- ps1.16S.R
otu_table(ps_cssBR) <- otu_table(css_countsBR, taxa_are_rows = TRUE)

ps_cssFR <- ps1.ITS.R
otu_table(ps_cssFR) <- otu_table(css_countsFR, taxa_are_rows = TRUE)


# Get relative abundance
ps1.16S.R_RA <- transform_sample_counts(ps_cssBR, function(x) 100 * x/sum(x))

ps1.ITS.R_RA <- transform_sample_counts(ps_cssFR, function(x) 100 * x/sum(x))

# Create dataframe from phyloseq object
dat_ps1.16S.R_RA <- psmelt(ps1.16S.R_RA)
dat_ps1.ITS.R_RA <- psmelt(ps1.ITS.R_RA)

####### Filter for tax assignment >98% confidence
dat_ps1.16S.R_RA <- subset(dat_ps1.16S.R_RA, confidence >= 0.98) 
dat_ps1.ITS.R_RA <- subset(dat_ps1.ITS.R_RA, confidence >= 0.98)

#Take the total RA sum of each genus 
RA.BAC.R <- aggregate(dat_ps1.16S.R_RA$Abundance, by = list(genus = dat_ps1.16S.R_RA$Genus), FUN=sum) 
RA.BAC.R <- dplyr::rename(RA.BAC.R, TotalRA = `x`)

RA.FUN.R <- aggregate(dat_ps1.ITS.R_RA$Abundance, by = list(genus = dat_ps1.ITS.R_RA$Genus), FUN=sum) 
RA.FUN.R <- dplyr::rename(RA.FUN.R, TotalRA = `x`)

#Determining 95th quantile of the RA data 
quantile.95.RA.B.R <- quantile(RA.BAC.R$TotalRA, 0.95)
quantile.95.RA.B.R

quantile.95.RA.F.R <- quantile(RA.FUN.R$TotalRA, 0.95)
quantile.95.RA.F.R

#Selecting for genera that fit the 95th cutoff
Bac.R.abund.95 <- subset(RA.BAC.R, TotalRA >= quantile.95.RA.B.R) 
Bac.R.abund.gen.95 <- Bac.R.abund.95$genus #make list of genera that fit this criteria

Fun.R.abund.95 <- subset(RA.FUN.R, TotalRA >= quantile.95.RA.F.R) 
Fun.R.abund.gen.95 <- Fun.R.abund.95$genus

```

## Determine the frequency cutoff - Roots
```{r, warning=FALSE, message=FALSE, echo=FALSE}

## To create ASVs count data frame from phyloseq object - agglomerated by Genus

# For Bacteria
ps1.16S.R_genus <- tax_glom(ps1.16S.R, "Genus")
asv.ps1.16S.R_genus <- otu_table(ps1.16S.R_genus)
df.ps1.16S.R_genus <- as.data.frame(asv.ps1.16S.R_genus)
identical(rownames(df.ps1.16S.R_genus), taxa_names(ps1.16S.R_genus)) #To verify if rownames are the same. If colnames needed just transpose 
rownames(df.ps1.16S.R_genus) <- as.data.frame(tax_table(ps1.16S.R_genus))$Genus  #Rename rownames using taxa (genus)

#For Fungi
ps1.ITS.R_genus <- tax_glom(ps1.ITS.R, "Genus")
asv.ps1.ITS.R_genus <- otu_table(ps1.ITS.R_genus)
df.ps1.ITS.R_genus <- as.data.frame(asv.ps1.ITS.R_genus)
identical(rownames(df.ps1.ITS.R_genus), taxa_names(ps1.ITS.R_genus)) #To verify if rownames are the same. If colnames needed just transpose 
rownames(df.ps1.ITS.R_genus) <- as.data.frame(tax_table(ps1.ITS.R_genus))$Genus  #Rename rownames using taxa (genus)

# To determine presence/abundance of genera in ASVs count table

PA.BAC.R <- 1*((df.ps1.16S.R_genus>0)==1)   #presence-absence data: 1 if present, 0 if not
Occ.BAC.R <- rowSums(PA.BAC.R)/ncol(PA.BAC.R) #Calculate occupancy
Freq.BAC.R <- add_rownames(as.data.frame(cbind(Occ.BAC.R)), 'genus') #creates new df with freq 

PA.FUN.R <- 1*((df.ps1.ITS.R_genus>0)==1) 
Occ.FUN.R <- rowSums(PA.FUN.R )/ncol(PA.FUN.R)
Freq.FUN.R <- add_rownames(as.data.frame(cbind(Occ.FUN.R)), 'genus') 

#Determining 95th quantile of the PA data 
quantile.95.PA.B.R <- quantile(Freq.BAC.R$Occ.BAC.R, 0.95)
quantile.95.PA.B.R

quantile.95.PA.F.R <- quantile(Freq.FUN.R$Occ.FUN.R, 0.95)
quantile.95.PA.F.R

#Selecting for genera that fit the 95th cutoff
Bac.R.freq.95 <- subset(Freq.BAC.R, Freq.BAC.R$Occ.BAC.R >= quantile.95.PA.B.R) 
Bac.R.freq.gen.95 <- Bac.R.freq.95$genus

Fun.R.freq.95 <- subset(Freq.FUN.R, Occ.FUN.R >= quantile.95.PA.F.R) #select cutoff 
Fun.R.freq.gen.95 <- Fun.R.freq.95$genus

```
## Subset the core from the data - Roots 
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Venn Diagram showing the shared genera number
Baclist.R.95 <- list(Frequency = Bac.R.freq.gen.95, Relative_abundance = Bac.R.abund.gen.95)
Bac.R.Venn <- ggvenn(Baclist.R.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 10) + 
            ggtitle('Bacteria in Roots')

Funlist.R.95 <- list(Frequency = Fun.R.freq.gen.95, Relative_abundance = Fun.R.abund.gen.95)

Fun.R.Venn <- ggvenn(Funlist.R.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 10) + 
            ggtitle('Fungi in Roots')

# Save venn diagrams
#Bacteria
ggsave("Rplots_survey/Bac.Roots.Venn.png", Bac.R.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#Fungi
ggsave("Rplots_survey/Fun.Roots.Venn.png", Fun.R.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#List of intersect -- This is our 95th percentile core (which overlaps with Shade and Stonisek 2019 analysis later in script)
Bac.R.Core.95 <- Reduce(intersect, Baclist.R.95)
Fun.R.Core.95 <- Reduce(intersect, Funlist.R.95)

BAC.R.Core <- subset_taxa(ps1.16S.R, Genus %in% c(Bac.R.Core.95))
FUN.R.Core <- subset_taxa(ps1.ITS.R, Genus %in% c(Fun.R.Core.95))


```

## Fix RA to be relative ONLY to each core genera , this is "across samples" Roots
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#BACTERIA
# Agglomerate taxa by genus level
BAC.R.Core_glomG <- tax_glom(BAC.R.Core, taxrank = 'Genus', NArm = FALSE)

# Get relative abundance in %
BAC.R.Core_RA <- transform_sample_counts(BAC.R.Core_glomG , function(x) 100 * x/sum(x))

#FUNGI
# Agglomerate taxa by genus level
FUN.R.Core_glomG <- tax_glom(FUN.R.Core, taxrank = 'Genus', NArm = FALSE)
# Get relative abundance in %
FUN.R.Core_RA <- transform_sample_counts(FUN.R.Core_glomG, function(x) 100 * x/sum(x))

#Transform to a data frame
dat_BAC.R.Core <- psmelt(BAC.R.Core_RA)
dat_FUN.R.Core <- psmelt(FUN.R.Core_RA)

```

## Roots Bacteria - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}
sample.type_levels.R <- c("Roots", "growthmedia")

tax_df <- dat_BAC.R.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.R <- dat_BAC.R.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels.R))
  #dplyr::filter(Abundance > 0)

genus.R_order <- abund_plot.R %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growthmedia'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreBL.colors <- c('#FB9A99', "goldenrod2", '#1F78B4') # "#E31A1C", '#1B9E77')  #"#E31A1C", '#66A61E', '#1B9E77'

#Final plot and save it

BBplot.16S.Rcore <- abund_plot.R %>%
  mutate(Genus = factor(Genus, levels = genus.R_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 15, 25)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreBL.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("CoreMicro_R_16S.png", BBplot.16S.Rcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

## Roots Fungi - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levelsF.R <- c("Roots", "growth.media")

taxF_df <- dat_FUN.R.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plotF.R <- dat_FUN.R.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(taxF_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(sample.type = factor(sample.type,
                                     levels = sample.type_levelsF.R))
  #dplyr::filter(Abundance > 0)

genusF.R_order <- abund_plotF.R %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growth.media'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreF.R.colors <- c("aquamarine", "plum1", "maroon2")
#Final plot and save it

BBplot.ITS.Rcore <- abund_plotF.R %>%
  mutate(Genus = factor(Genus, levels = genusF.R_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 15, 25)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreF.R.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("CoreMicro_R_ITS.png", BBplot.ITS.Rcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

# Growth media
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Sub-setting season 1 and 1 Pond from GCG
ps1.16S.G <- ps.16S.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "growthmedia")

ps1.ITS.G <- ps.ITS.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "growthmedia")


# Get relative abundance
ps1.16S.G_RA <- transform_sample_counts(ps1.16S.G, function(x) 100 * x/sum(x))

ps1.ITS.G_RA <- transform_sample_counts(ps1.ITS.G, function(x) 100 * x/sum(x))

# Create dataframe from phyloseq object
dat_ps1.16S.G_RA <- psmelt(ps1.16S.G_RA)
dat_ps1.ITS.G_RA <- psmelt(ps1.ITS.G_RA)

#Take the total RA sum of each genus 
RA.BAC.G <- aggregate(dat_ps1.16S.G_RA$Abundance, by = list(genus = dat_ps1.16S.G_RA$Genus), FUN=sum) 
RA.BAC.G <- dplyr::rename(RA.BAC.G, TotalRA = `x`)

RA.FUN.G <- aggregate(dat_ps1.ITS.G_RA$Abundance, by = list(genus = dat_ps1.ITS.G_RA$Genus), FUN=sum) 
RA.FUN.G <- dplyr::rename(RA.FUN.G, TotalRA = `x`)

#Determining 95th quantile of the RA data 
quantile.95.RA.B.G <- quantile(RA.BAC.G$TotalRA, 0.95)
quantile.95.RA.B.G

quantile.95.RA.F.G <- quantile(RA.FUN.G$TotalRA, 0.95)
quantile.95.RA.F.G

#Selecting for genera that fit the 95th cutoff
Bac.G.abund.95 <- subset(RA.BAC.G, TotalRA >= quantile.95.RA.B.G) 
Bac.G.abund.gen.95 <- Bac.G.abund.95$genus #make list of genera that fit this criteria

Fun.G.abund.95 <- subset(RA.FUN.G, TotalRA >= quantile.95.RA.F.G) 
Fun.G.abund.gen.95 <- Fun.G.abund.95$genus

```

## Determine the frequency cutoff - GM
```{r, warning=FALSE, message=FALSE, echo=FALSE}

## To create ASVs count data frame from phyloseq object - agglomerated by Genus

# For Bacteria
ps1.16S.G_genus <- tax_glom(ps1.16S.G, "Genus")
asv.ps1.16S.G_genus <- otu_table(ps1.16S.G_genus)
df.ps1.16S.G_genus <- as.data.frame(asv.ps1.16S.G_genus)
identical(rownames(df.ps1.16S.G_genus), taxa_names(ps1.16S.G_genus)) #To verify if rownames are the same. If colnames needed just transpose 
rownames(df.ps1.16S.G_genus) <- as.data.frame(tax_table(ps1.16S.G_genus))$Genus  #Rename rownames using taxa (genus)

#For Fungi
ps1.ITS.G_genus <- tax_glom(ps1.ITS.G, "Genus")
asv.ps1.ITS.G_genus <- otu_table(ps1.ITS.G_genus)
df.ps1.ITS.G_genus <- as.data.frame(asv.ps1.ITS.G_genus)
identical(rownames(df.ps1.ITS.G_genus), taxa_names(ps1.ITS.G_genus)) #To verify if rownames are the same. If colnames needed just transpose 
rownames(df.ps1.ITS.G_genus) <- as.data.frame(tax_table(ps1.ITS.G_genus))$Genus  #Rename rownames using taxa (genus)

# To determine presence/abundance of genera in ASVs count table

PA.BAC.G <- 1*((df.ps1.16S.G_genus>0)==1)   #presence-absence data: 1 if present, 0 if not
Occ.BAC.G <- rowSums(PA.BAC.G)/ncol(PA.BAC.G) #Calculate occupancy
Freq.BAC.G <- add_rownames(as.data.frame(cbind(Occ.BAC.G)), 'genus') #creates new df with freq 

PA.FUN.G <- 1*((df.ps1.ITS.G_genus>0)==1) 
Occ.FUN.G <- rowSums(PA.FUN.G )/ncol(PA.FUN.G)
Freq.FUN.G <- add_rownames(as.data.frame(cbind(Occ.FUN.G)), 'genus') 

#Determining 95th quantile of the PA data 
quantile.95.PA.B.G <- quantile(Freq.BAC.G$Occ.BAC.G, 0.95)
quantile.95.PA.B.G

quantile.95.PA.F.G <- quantile(Freq.FUN.G$Occ.FUN.G, 0.95)
quantile.95.PA.F.G

#Selecting for genera that fit the 95th cutoff
Bac.G.freq.95 <- subset(Freq.BAC.G, Freq.BAC.G$Occ.BAC.G >= quantile.95.PA.B.G) 
Bac.G.freq.gen.95 <- Bac.G.freq.95$genus

Fun.G.freq.95 <- subset(Freq.FUN.G, Occ.FUN.G >= quantile.95.PA.F.G) #select cutoff 
Fun.G.freq.gen.95 <- Fun.G.freq.95$genus

```

## Subset the core from the data GM 
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Venn Diagram showing the shared genera number
Baclist.G.95 <- list(Frequency = Bac.G.freq.gen.95, Relative_abundance = Bac.G.abund.gen.95)
Bac.G.Venn <- ggvenn(Baclist.G.95,
            stroke_size = 0.75, set_name_size = 7, text_size = 9) + 
            ggtitle('Bacteria in Growth media')

Funlist.G.95 <- list(Frequency = Fun.G.freq.gen.95, Relative_abundance = Fun.G.abund.gen.95)

Fun.G.Venn <- ggvenn(Funlist.G.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 10) + 
            ggtitle('Fungi in Growth media')

# Save venn diagrams
#Bacteria
ggsave("Rplots_survey/Bac.GM.Venn.png", Bac.G.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#Fungi
ggsave("Rplots_survey/Fun.GM.Venn.png", Fun.G.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)


#List of intersect -- This is our 95th percentile core (which overlaps with Shade and Stonisek 2019 analysis later in script)
Bac.G.Core.95 <- Reduce(intersect, Baclist.G.95)
Fun.G.Core.95 <- Reduce(intersect, Funlist.G.95)

BAC.G.Core <- subset_taxa(ps1.16S.G, Genus %in% c(Bac.G.Core.95))
FUN.G.Core <- subset_taxa(ps1.ITS.G, Genus %in% c(Fun.G.Core.95))

```

## Fix RA to be relative ONLY to each core genera , this is "across samples" GM 
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#BACTERIA
# Agglomerate taxa by genus level
BAC.G.Core_glomG <- tax_glom(BAC.G.Core, taxrank = 'Genus', NArm = FALSE)

# Get relative abundance in %
BAC.G.Core_RA <- transform_sample_counts(BAC.G.Core_glomG , function(x) 100 * x/sum(x))

#FUNGI
# Agglomerate taxa by genus level
FUN.G.Core_glomG <- tax_glom(FUN.G.Core, taxrank = 'Genus', NArm = FALSE)
# Get relative abundance in %
FUN.G.Core_RA <- transform_sample_counts(FUN.G.Core_glomG, function(x) 100 * x/sum(x))

#Transform to a data frame
dat_BAC.G.Core <- psmelt(BAC.G.Core_RA)
dat_FUN.G.Core <- psmelt(FUN.G.Core_RA)

```

## GM Bacteria - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}
sample.type_levels.G <- c("growthmedia", "Roots")

tax_df <- dat_BAC.G.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.G <- dat_BAC.G.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels.G))
  #dplyr::filter(Abundance > 0)

genus.G_order <- abund_plot.G %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'growthmedia'="Growth media",
  'Roots'="Roots")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreBL.colors <- c("goldenrod2", '#66A61E', '#1F78B4',"firebrick4")

#Final plot and save it

BBplot.16S.Gcore <- abund_plot.G %>%
  mutate(Genus = factor(Genus, levels = genus.G_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 10, 20)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreBL.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("Rplots_survey/CoreMicro_G_16S.png", BBplot.16S.Gcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

## GM Fungi - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levelsF.G <- c("Roots", "growthmedia")

taxF_df <- dat_FUN.G.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plotF.G <- dat_FUN.G.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(taxF_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(sample.type = factor(sample.type,
                                     levels = sample.type_levelsF.G))
  #dplyr::filter(Abundance > 0)

genusF.G_order <- abund_plotF.G %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growthmedia'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreF.G.colors <- c("aquamarine", "mediumorchid", "plum1", "steelblue3", "maroon2")
#Final plot and save it

BBplot.ITS.Gcore <- abund_plotF.G %>%
  mutate(Genus = factor(Genus, levels = genusF.G_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 20, 45)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreF.G.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)), panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("Rplots_survey/CoreMicro_G_ITS.png", BBplot.ITS.Gcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

# Leaves
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Sub-setting season 1 and 1 Pond from GCG
ps1.16S.L <- ps.16S.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "Leaves")

ps1.ITS.L <- ps.ITS.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "Leaves")

# Get relative abundance
ps1.16S.L_RA <- transform_sample_counts(ps1.16S.L, function(x) 100 * x/sum(x))

ps1.ITS.L_RA <- transform_sample_counts(ps1.ITS.L, function(x) 100 * x/sum(x))

# Create dataframe from phyloseq object
dat_ps1.16S.L_RA <- psmelt(ps1.16S.L_RA)
dat_ps1.ITS.L_RA <- psmelt(ps1.ITS.L_RA)

#Take the total RA sum of each genus 
RA.BAC.L <- aggregate(dat_ps1.16S.L_RA$Abundance, by = list(genus = dat_ps1.16S.L_RA$Genus), FUN=sum) 
RA.BAC.L <- dplyr::rename(RA.BAC.L, TotalRA = `x`)

RA.FUN.L <- aggregate(dat_ps1.ITS.L_RA$Abundance, by = list(genus = dat_ps1.ITS.L_RA$Genus), FUN=sum) 
RA.FUN.L <- dplyr::rename(RA.FUN.L, TotalRA = `x`)

#Determining 95th quantile of the RA data 
quantile.95.RA.B.L <- quantile(RA.BAC.L$TotalRA, 0.95)
quantile.95.RA.B.L

quantile.95.RA.F.L <- quantile(RA.FUN.L$TotalRA, 0.95)
quantile.95.RA.F.L

#Selecting for genera that fit the 95th cutoff
Bac.L.abund.95 <- subset(RA.BAC.L, TotalRA >= quantile.95.RA.B.L) 
Bac.L.abund.gen.95 <- Bac.L.abund.95$genus #make list of genera that fit this criteria

Fun.L.abund.95 <- subset(RA.FUN.L, TotalRA >= quantile.95.RA.F.L) 
Fun.L.abund.gen.95 <- Fun.L.abund.95$genus

```

## Determine the frequency cutoff - Leaves 
```{r, warning=FALSE, message=FALSE, echo=FALSE}

## To create ASVs count data frame from phyloseq object - agglomerated by Genus

# For Bacteria
ps1.16S.L_genus <- tax_glom(ps1.16S.L, "Genus")
asv.ps1.16S.L_genus <- otu_table(ps1.16S.L_genus)
df.ps1.16S.L_genus <- as.data.frame(asv.ps1.16S.L_genus)
identical(rownames(df.ps1.16S.L_genus), taxa_names(ps1.16S.L_genus)) #To verify if rownames are the same. If colnames needed just transpose 
rownames(df.ps1.16S.L_genus) <- as.data.frame(tax_table(ps1.16S.L_genus))$Genus  #Rename rownames using taxa (genus)

#For Fungi
ps1.ITS.L_genus <- tax_glom(ps1.ITS.L, "Genus")
asv.ps1.ITS.L_genus <- otu_table(ps1.ITS.L_genus)
df.ps1.ITS.L_genus <- as.data.frame(asv.ps1.ITS.L_genus)
identical(rownames(df.ps1.ITS.L_genus), taxa_names(ps1.ITS.L_genus)) #To verify if rownames are the same. If colnames needed just transpose 
rownames(df.ps1.ITS.L_genus) <- as.data.frame(tax_table(ps1.ITS.L_genus))$Genus  #Rename rownames using taxa (genus)

# To determine presence/abundance of genera in ASVs count table

PA.BAC.L <- 1*((df.ps1.16S.L_genus>0)==1)   #presence-absence data: 1 if present, 0 if not
Occ.BAC.L <- rowSums(PA.BAC.L)/ncol(PA.BAC.L) #Calculate occupancy
Freq.BAC.L <- add_rownames(as.data.frame(cbind(Occ.BAC.L)), 'genus') #creates new df with freq 

PA.FUN.L <- 1*((df.ps1.ITS.L_genus>0)==1) 
Occ.FUN.L <- rowSums(PA.FUN.L )/ncol(PA.FUN.L)
Freq.FUN.L <- add_rownames(as.data.frame(cbind(Occ.FUN.L)), 'genus') 

#Determining 95th quantile of the PA data 
quantile.95.PA.B.L <- quantile(Freq.BAC.L$Occ.BAC.L, 0.95)
quantile.95.PA.B.L

#To see the distribution
#hist(Freq.BAC.L$Occ.BAC.L)

quantile.95.PA.F.L <- quantile(Freq.FUN.L$Occ.FUN.L, 0.95)
quantile.95.PA.F.L

#Selecting for genera that fit the 95th cutoff
Bac.L.freq.95 <- subset(Freq.BAC.L, Freq.BAC.L$Occ.BAC.L >= quantile.95.PA.B.L) 
Bac.L.freq.gen.95 <- Bac.L.freq.95$genus

Fun.L.freq.95 <- subset(Freq.FUN.L, Occ.FUN.L >= quantile.95.PA.F.L) #select cutoff 
Fun.L.freq.gen.95 <- Fun.L.freq.95$genus

```

## Subset the core from the data Leaves 
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Venn Diagram showing the shared genera number
Baclist.L.95 <- list(Frequency = Bac.L.freq.gen.95, Relative_abundance = Bac.L.abund.gen.95)
Bac.L.Venn <- ggvenn(Baclist.L.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 9) + 
            ggtitle('Bacteria in Leaves')

Funlist.L.95 <- list(Frequency = Fun.L.freq.gen.95, Relative_abundance = Fun.L.abund.gen.95)

Fun.L.Venn <- ggvenn(Funlist.L.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 9) + 
            ggtitle('Fungi in Leaves')

# Save venn diagrams
#Bacteria
ggsave("Rplots_survey/Bac.Leaves.Venn.png", Bac.L.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#Fungi
ggsave("Rplots_survey/Fun.Leaves.Venn.png", Fun.L.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)


#List of intersect -- This is our 95th percentile core (which overlaps with Shade and Stonisek 2019 analysis later in script)
Bac.L.Core.95 <- Reduce(intersect, Baclist.L.95)
Fun.L.Core.95 <- Reduce(intersect, Funlist.L.95)

BAC.L.Core <- subset_taxa(ps1.16S.L, Genus %in% c(Bac.L.Core.95))
FUN.L.Core <- subset_taxa(ps1.ITS.L, Genus %in% c(Fun.L.Core.95))


```
## Fix RA to be relative ONLY to each core genera , this is "across samples" Leaves
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#BACTERIA
# Agglomerate taxa by genus level
BAC.L.Core_glomG <- tax_glom(BAC.L.Core, taxrank = 'Genus', NArm = FALSE)

# Get relative abundance in %
BAC.L.Core_RA <- transform_sample_counts(BAC.L.Core_glomG , function(x) 100 * x/sum(x))

#FUNGI
# Agglomerate taxa by genus level
FUN.L.Core_glomG <- tax_glom(FUN.L.Core, taxrank = 'Genus', NArm = FALSE)
# Get relative abundance in %
FUN.L.Core_RA <- transform_sample_counts(FUN.L.Core_glomG, function(x) 100 * x/sum(x))

#Transform to a data frame
dat_BAC.L.Core <- psmelt(BAC.L.Core_RA)
dat_FUN.L.Core <- psmelt(FUN.L.Core_RA)

```

## Leaves Bacteria - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levels.L <- c("Leaves", "growthmedia")

tax_df <- dat_BAC.L.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.L <- dat_BAC.L.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Leptolyngbya", Genus),
                               "Leptolyngbya", Genus)) %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels.L))


genus.L_order <- abund_plot.L %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Leaves'="Leaves",
  'growthmedia'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}


#Color pallete
coreB.colors <- c("goldenrod2", '#80CDC1', '#1F78B4')
  
#Final plot and save it

BBplot.16S.Lcore <- abund_plot.L %>%
  mutate(Genus = factor(Genus, levels = genus.L_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 15, 25)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreB.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("Rplots_survey/CoreMicro_L_16S.png", BBplot.16S.Lcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

## Leaves Fungi - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levels.L <- c("Leaves", "growthmedia")

taxF_df <- dat_FUN.L.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plotF.L <- dat_FUN.L.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(taxF_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(sample.type = factor(sample.type,
                                     levels = sample.type_levels.L))

genusF.L_order <- abund_plotF.L %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Leaves'="Leaves",
  'growthmedia'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreF.colors <- c("aquamarine","mediumorchid", "maroon2")
  
#Final plot and save it

BBplot.ITS.Lcore <- abund_plotF.L %>%
  mutate(Genus = factor(Genus, levels = genusF.L_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 20, 45)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreF.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("Rplots_survey/CoreMicro_L_ITS.png", BBplot.ITS.Lcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

# Nutrient solution
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Sub-setting season 1 and 1 Pond from GCG
ps1.16S.W <- ps.16S.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "Filtered.water", sampling.point != "Water.source")

ps1.ITS.W <- ps.ITS.nc %>% ps_filter(comparison == "only") %>%
    ps_filter(sample.type == "Filtered.water", sampling.point != "Water.source")

# Get relative abundance
ps1.16S.W_RA <- transform_sample_counts(ps1.16S.W, function(x) 100 * x/sum(x))

ps1.ITS.W_RA <- transform_sample_counts(ps1.ITS.W, function(x) 100 * x/sum(x))

# Create dataframe from phyloseq object
dat_ps1.16S.W_RA <- psmelt(ps1.16S.W_RA)
dat_ps1.ITS.W_RA <- psmelt(ps1.ITS.W_RA)

#Take the total RA sum of each genus 
RA.BAC.W <- aggregate(dat_ps1.16S.W_RA$Abundance, by = list(genus = dat_ps1.16S.W_RA$Genus), FUN=sum) 
RA.BAC.W <- dplyr::rename(RA.BAC.W, TotalRA = `x`)

RA.FUN.W <- aggregate(dat_ps1.ITS.W_RA$Abundance, by = list(genus = dat_ps1.ITS.W_RA$Genus), FUN=sum) 
RA.FUN.W <- dplyr::rename(RA.FUN.W, TotalRA = `x`)

#Determining 90th quantile of the RA data 
quantile.95.RA.B.W <- quantile(RA.BAC.W$TotalRA, 0.95)
quantile.95.RA.B.W

quantile.95.RA.F.W <- quantile(RA.FUN.W$TotalRA, 0.95)
quantile.95.RA.F.W

#Selecting for genera that fit the 90th cutoff
Bac.W.abund.95 <- subset(RA.BAC.W, TotalRA >= quantile.95.RA.B.W) 
Bac.W.abund.gen.95 <- Bac.W.abund.95$genus #make list of genera that fit this criteria

Fun.W.abund.95 <- subset(RA.FUN.W, TotalRA >= quantile.95.RA.F.W) 
Fun.W.abund.gen.95 <- Fun.W.abund.95$genus

```

## Determine the frequency cutoff - Nutrient sol. 
```{r, warning=FALSE, message=FALSE, echo=FALSE}

## To create ASVs count data frame from phyloseq object - agglomerated by Genus

# For Bacteria
ps1.16S.W_genus <- tax_glom(ps1.16S.W, "Genus")
asv.ps1.16S.W_genus <- otu_table(ps1.16S.W_genus)
df.ps1.16S.W_genus <- as.data.frame(asv.ps1.16S.W_genus)
identical(rownames(df.ps1.16S.W_genus), taxa_names(ps1.16S.W_genus)) #To verify if rownames are the same. If colnames needed just transpose 

#Problem with duplicate rownames due to duplicate Genus names (Incertae Sedis)
#b <-  data.frame(tax_table(ps1.16S.W_genus)) #Rename rownames using taxa (genus)

#write.table(b, file="taxanames.16SWater.txt")

c <- read.table("taxanames.16SWater.txt")

d <- c$Genus

#Rename rows
row.names(df.ps1.16S.W_genus) = d

#For Fungi
ps1.ITS.W_genus <- tax_glom(ps1.ITS.W, "Genus")
asv.ps1.ITS.W_genus <- otu_table(ps1.ITS.W_genus)
df.ps1.ITS.W_genus <- as.data.frame(asv.ps1.ITS.W_genus)
identical(rownames(df.ps1.ITS.W_genus), taxa_names(ps1.ITS.W_genus)) #To verify if rownames are the same. If colnames needed just transpose 

rownames(df.ps1.ITS.W_genus) <- as.data.frame(tax_table(ps1.ITS.W_genus))$Genus  #Rename rownames using taxa (genus)

# To determine presence/abundance of genera in ASVs count table
PA.BAC.W <- 1*((df.ps1.16S.W_genus>0)==1)   #presence-absence data: 1 if present, 0 if not
Occ.BAC.W <- rowSums(PA.BAC.W)/ncol(PA.BAC.W) #Calculate occupancy
Freq.BAC.W <- add_rownames(as.data.frame(cbind(Occ.BAC.W)), 'genus') #creates new df with freq 

PA.FUN.W <- 1*((df.ps1.ITS.W_genus>0)==1) 
Occ.FUN.W <- rowSums(PA.FUN.W )/ncol(PA.FUN.W)
Freq.FUN.W <- add_rownames(as.data.frame(cbind(Occ.FUN.W)), 'genus') 

#Determining 95th quantile of the PA data 
quantile.95.PA.B.W <- quantile(Freq.BAC.W$Occ.BAC.W, 0.95)
quantile.95.PA.B.W

quantile.95.PA.F.W <- quantile(Freq.FUN.W$Occ.FUN.W, 0.95)
quantile.95.PA.F.W

#Selecting for genera that fit the 90th cutoff
Bac.W.freq.95 <- subset(Freq.BAC.W, Freq.BAC.W$Occ.BAC.W >= quantile.95.PA.B.W) 
Bac.W.freq.gen.95 <- Bac.W.freq.95$genus

Fun.W.freq.95 <- subset(Freq.FUN.W, Occ.FUN.W >= quantile.95.PA.F.W) #select cutoff 
Fun.W.freq.gen.95 <- Fun.W.freq.95$genus

```

## Subset the core from the data Nutrient sol.
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Venn Diagram showing the shared genera number
Baclist.W.95 <- list(Frequency = Bac.W.freq.gen.95, Relative_abundance = Bac.W.abund.gen.95)
Bac.W.Venn <- ggvenn(Baclist.W.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 7) + 
            ggtitle('Bacteria in Nutrient solution')

Funlist.W.95 <- list(Frequency = Fun.W.freq.gen.95, Relative_abundance = Fun.W.abund.gen.95)

Fun.W.Venn <- ggvenn(Funlist.W.95,
            stroke_size = 0.5, set_name_size = 7, text_size = 7) + 
            ggtitle('Fungi in Nutrient solution')

# Save venn diagrams
#Bacteria
ggsave("Rplots_survey/Bac.Water.Venn.png", Bac.W.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#Fungi
ggsave("Rplots_survey/Fun.Water.Venn.png", Fun.W.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#List of intersect -- This is our 95th percentile core in nutrient solution (water source removed)
Bac.W.Core.95 <- Reduce(intersect, Baclist.W.95) 
Fun.W.Core.95 <- Reduce(intersect, Funlist.W.95) 

BAC.W.Core <- subset_taxa(ps1.16S.W, Genus %in% c(Bac.W.Core.95))
FUN.W.Core <- subset_taxa(ps1.ITS.W, Genus %in% c(Fun.W.Core.95))


```
## Fix RA to be relative ONLY to each core genera , this is "across samples" Nutrient solution
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#BACTERIA
# Agglomerate taxa by genus level
BAC.W.Core_glomG <- tax_glom(BAC.W.Core, taxrank = 'Genus', NArm = FALSE)

# Get relative abundance in %
BAC.W.Core_RA <- transform_sample_counts(BAC.W.Core_glomG , function(x) 100 * x/sum(x))

#FUNGI
# Agglomerate taxa by genus level
FUN.W.Core_glomG <- tax_glom(FUN.W.Core, taxrank = 'Genus', NArm = FALSE)
# Get relative abundance in %
FUN.W.Core_RA <- transform_sample_counts(FUN.W.Core_glomG, function(x) 100 * x/sum(x))

#Transform to a data frame
dat_BAC.W.Core <- psmelt(BAC.W.Core_RA)
dat_FUN.W.Core <- psmelt(FUN.W.Core_RA)

```

## Nut.sol Bacteria - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}
sample.type_levels.W <- c("Filtered.water", "growthmedia")

tax_df <- dat_BAC.W.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.W <- dat_BAC.W.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels.W))
  #dplyr::filter(Abundance > 0)

genus.W_order <- abund_plot.W %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Filtered.water'="Nutrient\nsolution",
  'growthmedia'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreBL.colors <- c('#FB9A99', "goldenrod2", "coral1", '#1F78B4')

#Final plot and save it

BBplot.16S.Wcore <- abund_plot.W %>%
  mutate(Genus = factor(Genus, levels = genus.W_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(1, 5, 15)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreBL.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("Rplots_survey/CoreMicro_W_16S.png", BBplot.16S.Wcore, 
  width = 9,
  height = 12,
  units = "cm",
  dpi = 600)

```

## Nut.sol Fungi - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levelsF.W <- c("Filtered.water", "growthmedia")

taxF_df <- dat_FUN.W.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plotF.W <- dat_FUN.W.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(taxF_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(sample.type = factor(sample.type,
                                     levels = sample.type_levelsF.W))
  #dplyr::filter(Abundance > 0)

genusF.W_order <- abund_plotF.W %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Filtered.water'="Nutrient\nsolution",
  'growthmedia'="Growth media")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreF.W.colors <- c("aquamarine", "mediumorchid", "plum1", "maroon2")
#Final plot and save it

BBplot.ITS.Wcore <- abund_plotF.W %>%
  mutate(Genus = factor(Genus, levels = genusF.W_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(5, 20, 45)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreF.W.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom",
        legend.location = "plot") + guides(fill = "none")

ggsave("Rplots_survey/CoreMicro_W_ITS.png", BBplot.ITS.Wcore, 
  width = 8,
  height = 12,
  units = "cm",
  dpi = 600)

```

```{r, warning=FALSE, message=FALSE, echo=FALSE}

##### Final list CORE TAXA

#Make a list with Core taxa from all sample types
Baclist.95.all <- list(Roots = Bac.R.Core.95, Growth_media = Bac.G.Core.95, NutrienSol = Bac.W.Core.95, Leaves = Bac.L.Core.95)
Funlist.95.all <- list(Roots = Fun.R.Core.95, Growth_media = Fun.G.Core.95, NutrienSol = Fun.W.Core.95, Leaves = Fun.L.Core.95)

#To get a vector with all core taxa
Bactaxa.95.all <- c(Bac.R.Core.95, Bac.G.Core.95, Bac.W.Core.95, Bac.L.Core.95)
Funtaxa.95.all <- c(Fun.R.Core.95, Fun.G.Core.95, Fun.W.Core.95, Fun.L.Core.95)

#Get the shared ASVs in the list
Final.Bac.core <- Reduce(intersect, Baclist.95.all)
Final.Fun.core <- Reduce(intersect, Funlist.95.all)

#Subset the taxa from the whole dataset
#Bacteria
ps1.16S <- ps.16S.nc %>% ps_filter(sampling.season == "1") %>%
  ps_filter(comparison == "only") %>%
    ps_filter(source == c("Plant", "Water"))

BAC.final.Core <- subset_taxa(ps1.16S, Genus %in% c(Final.Bac.core))

#Fungi
ps1.ITS <- ps.ITS.nc %>% ps_filter(sampling.season == "1") %>%
  ps_filter(comparison == "only") %>%
    ps_filter(source == c("Plant", "Water"))

FUN.final.Core <- subset_taxa(ps1.ITS, Genus %in% c(Final.Fun.core))

############# Fix RA to be relative ONLY to each core genera   ####REALLY NEEDED???
##BACTERIA
## Agglomerate taxa by genus level
BAC.final.Core_glomG <- tax_glom(BAC.final.Core, taxrank = 'Genus', NArm = FALSE)

## Get relative abundance in %
BAC.final.Core_RA <- transform_sample_counts(BAC.final.Core_glomG , function(x) 100 * x/sum(x))

##FUNGI
## Agglomerate taxa by genus level
FUN.final.Core_glomG <- tax_glom(FUN.final.Core, taxrank = 'Genus', NArm = FALSE)
## Get relative abundance in %
FUN.final.Core_RA <- transform_sample_counts(FUN.final.Core_glomG, function(x) 100 * x/sum(x))

#Transform to a data frame
dat_BAC.final.Core <- psmelt(BAC.final.Core_RA)
dat_FUN.final.Core <- psmelt(FUN.final.Core_RA)

#dat_BAC.final.Core <- psmelt(BAC.final.Core)
#dat_FUN.final.Core <- psmelt(FUN.final.Core)

#Venn Diagram showing the shared genera number

library(RColorBrewer)
myColors <- c("#A6761D", "#FDBF6F", "#1F78B4", "#33A02C")

Bac.all.Venn <- ggvenn(Baclist.95.all,
            stroke_size = 0.5, set_name_size = 7, text_size = 5, fill_color = myColors) + 
            ggtitle('Bacteria in Roots, GM, Nut.Sol and Leaves')

Fun.all.Venn <- ggvenn(Funlist.95.all,
            stroke_size = 0.5, set_name_size = 7, text_size = 5, fill_color = myColors) + 
            ggtitle('Fungi in in Roots, GM, Nut.Sol and Leaves')

# Save venn diagrams
#Bacteria
ggsave("Rplots_survey/Bac.all.Venn.png", Bac.all.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

#Fungi
ggsave("Rplots_survey/Fun.all.Venn.png", Fun.all.Venn, 
  width = 40,
  height = 20,
  units = "cm",
  dpi = 600)

```


## Final Bacteria - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levels <- c("Roots", "growthmedia", "Filtered.water", "Leaves")

tax_df <- dat_BAC.final.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot <- dat_BAC.final.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels))

  #dplyr::filter(Abundance > 0)

genus_order <- abund_plot %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growthmedia'="Growth\nmedia",
  'Filtered.water' = "Nutrient\nsolution",
  'Leaves' = "Leaves")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreBL.colors <- c('goldenrod2', '#1F78B4')

#Final plot and save it

BBplot.16S.core <- abund_plot %>%
  mutate(Genus = factor(Genus, levels = genus_order)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(1, 5, 15, 25, 45, 60)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreBL.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom")

ggsave("Rplots_survey/CoreMicro.16S.png", BBplot.16S.core, 
  width = 20,
  height = 6,
  units = "cm",
  dpi = 600)

```

## Final Fungi - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levels <- c("Roots", "growthmedia", "Filtered.water", "Leaves")

tax_df <- dat_FUN.final.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.F <- dat_FUN.final.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels))

  #dplyr::filter(Abundance > 0)

genus_order.F <- abund_plot.F %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growthmedia'="Growth\nmedia",
  'Filtered.water' = "Nutrient\nsolution",
  'Leaves' = "Leaves")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreBL.colors <- c("aquamarine", "maroon2")

#Final plot and save it

BBplot.ITS.core <- abund_plot.F %>%
  mutate(Genus = factor(Genus, levels = genus_order.F)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(1, 5, 15, 35, 65)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreBL.colors) +
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank())

ggsave("Rplots_survey/CoreMicro.ITS.png", BBplot.ITS.core, 
  width = 20,
  height = 6,
  units = "cm",
  dpi = 600)

```

# Get data to plot all core taxa
```{r, warning=FALSE, message=FALSE, echo=FALSE}
###Make a list with Core taxa from all sample types
#Bacteria
Baclist.95.all <- list(Roots = Bac.R.Core.95, Growth_media = Bac.G.Core.95, NutrienSol = Bac.W.Core.95, Leaves = Bac.L.Core.95)
#Fungi
Funlist.95.all <- list(Roots = Fun.R.Core.95, Growth_media = Fun.G.Core.95, NutrienSol = Fun.W.Core.95, Leaves = Fun.L.Core.95)

#To get a vector with all core taxa
Bactaxa.95.all <- c(Bac.R.Core.95, Bac.G.Core.95, Bac.W.Core.95, Bac.L.Core.95)
Funtaxa.95.all <- c(Fun.R.Core.95, Fun.G.Core.95, Fun.W.Core.95, Fun.L.Core.95)

#Sub-setting
ps1.16S <- ps.16S.nc %>% ps_filter(source == c("Plant", "Water"), comparison == "only", sampling.point != "Water.source")

ps1.ITS <- ps.ITS.nc %>% ps_filter(source == c("Plant", "Water"), comparison == "only", sampling.point != "Water.source")


#To get all core taxa not only shared taxa (intersect)
BAC.finaltax.Core <- subset_taxa(ps1.16S, Genus %in% c(Bactaxa.95.all))
FUN.finaltax.Core <- subset_taxa(ps1.ITS, Genus %in% c(Funtaxa.95.all))

### Save the final phyloseq objects
#saveRDS(BAC.finaltax.Core, 'ps.Core16S.new.rds')
#saveRDS(FUN.finaltax.Core, 'ps.CoreITS.new.rds')

##Fix RA to be relative ONLY to each core genera  
##Bacteria
## Agglomerate taxa by genus level
BAC.finaltax.Core_glomG <- tax_glom(BAC.finaltax.Core, taxrank = 'Genus', NArm = FALSE)

## Get relative abundance in %
BAC.finaltax.Core_RA <- transform_sample_counts(BAC.finaltax.Core_glomG , function(x) 100 * x/sum(x))

##Fungi
## Agglomerate taxa by genus level
FUN.finaltax.Core_glomG <- tax_glom(FUN.finaltax.Core, taxrank = 'Genus', NArm = FALSE)
## Get relative abundance in %
FUN.finaltax.Core_RA <- transform_sample_counts(FUN.finaltax.Core_glomG, function(x) 100 * x/sum(x))

#Transform to a data frame
dat_BAC.finaltax.Core <- psmelt(BAC.finaltax.Core_RA)
dat_FUN.finaltax.Core <- psmelt(FUN.finaltax.Core_RA)

```

### Finaltax Bacteria - Bubble plot - need to modify color pallete !!
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levels <- c("Roots", "growthmedia", "Filtered.water", "Leaves")

tax_df <- dat_BAC.finaltax.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.tax <- dat_BAC.finaltax.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus)) %>%
  dplyr::mutate(Genus = ifelse(grepl("Leptolyngbya", Genus),
                               "Leptolyngbya", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels))

  #dplyr::filter(Abundance > 0)

genus_order.tax <- abund_plot.tax %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growthmedia'="Growth media",
  'Filtered.water' = "Nutrient\nsolution",
  'Leaves' = "Leaves")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreBL.colors <- c('#FB9A99', "goldenrod2",  '#80CDC1', "coral1", '#66A61E', '#1F78B4',"firebrick4")

#Final plot and save it

BBplot.16Stax.core <- abund_plot.tax %>%
  mutate(Genus = factor(Genus, levels = genus_order.tax)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(0, 1, 5, 10, 15)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreBL.colors)+
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom")

ggsave("Rplots_survey/CoreMicro.16Stax.png", BBplot.16Stax.core, 
  width = 20,
  height = 20,
  units = "cm",
  dpi = 600)

```

### Final Fungi - Bubble plot
```{r, warning=FALSE, message=FALSE, echo=FALSE}

sample.type_levels <- c("Roots", "growthmedia", "Filtered.water", "Leaves")

tax_df <- dat_FUN.finaltax.Core %>%
  select(Domain, Kingdom, Phylum, Class, Order, Family, Genus)

abund_plot.F.tax <- dat_FUN.finaltax.Core %>%
  dplyr::group_by(Genus, sample.type, system.type) %>% 
  dplyr::summarize(Abundance = mean(Abundance)) %>%
  dplyr::left_join(tax_df, by = "Genus", multiple = "all") %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus),
                sample.type = factor(sample.type,
                                     levels = sample.type_levels))

  #dplyr::filter(Abundance > 0)

genus_order.F.tax <- abund_plot.F.tax %>%
  dplyr::group_by(Genus) %>%
  dplyr::summarize(abund = mean(Abundance)) %>%
  arrange(desc(abund)) %>%
  pull(Genus)

#labels for facet_wrap
sample_names <- list(
  'Roots'="Roots",
  'growthmedia'="Growth media",
  'Filtered.water' = "Nutrient\nsolution",
  'Leaves' = "Leaves")

sample_labeller <- function(variable,value){
  return(sample_names[value])
}

#Color pallete
coreF.colors <- c("aquamarine", "mediumorchid", "plum1", "steelblue3", "maroon2")


#Final plot and save it

BBplot.ITStax.core <- abund_plot.F.tax %>%
  mutate(Genus = factor(Genus, levels = genus_order.F.tax)) %>% 
  ggplot() +
  aes(y = Genus, x = system.type,
      size = Abundance, fill = Phylum) +
  geom_point(shape = 21, color = "grey30") +
  facet_wrap(vars(sample.type), nrow = 1, labeller=sample_labeller) +
  scale_size(range = c(0, 8), name="Relative\nAbundance (%)",
             breaks = c(0, 5, 15, 30)) +
  scale_y_discrete(limits = rev) +
  scale_fill_manual(values = coreF.colors)+
  scale_x_discrete(name ="System type", 
                    limits=c("DWC","NFT","VD", "EF")) +
                    theme_few() +
                    theme(axis.text.x = element_text(angle = 90, margin=margin(r=0.2)),
                          axis.text.y = element_text(face = "italic", margin=margin(r=0.8)),
                          panel.grid = element_blank()) + theme(legend.position="bottom")

ggsave("CoreMicro.ITStax.png", BBplot.ITStax.core, 
  width = 32,
  height = 14,
  units = "cm",
  dpi = 600)

```

# Correlation Core abundance with env variables
```{r}
## Get relative abundance
BAC.finaltax.Core_RA <- transform_sample_counts(BAC.finaltax.Core_glomG , function(x) x/sum(x)) 

#Transform to a data frame
dat_BAC.finaltax.Core <- psmelt(BAC.finaltax.Core_RA)
dat_FUN.finaltax.Core <- psmelt(FUN.finaltax.Core_RA)

# Rearrage dataframe and Re-code water source variable
dat_BAC <- dat_BAC.finaltax.Core %>%
  dplyr::mutate(Genus = ifelse(grepl("Rhizobium", Genus),
                               "Rhizobium group", Genus)) %>% pivot_wider(names_from = Genus, values_from = Abundance) %>% dplyr::mutate(water.source = as.numeric(water.source == "Municipal"), water.source)

dat_FUN <- dat_FUN.finaltax.Core %>% dplyr::mutate(water.source = as.numeric(water.source == "Municipal"), water.source)

# Get matrix with variables of interest
variables.B <- dat_BAC[,c(12, 15, 19:21, 23, 31:85)]

variables.F <- dat_FUN.finaltax.Core[,c(3, 13, 16, 20, 21, 22, 24)]


# Pearson correlation (for continuous variables)
corr.df.B <- round(cor(variables.B, method = "pearson"), 2)

corr.df.F <- cor(variables.F, method = "pearson") 

# Test and visualize p-value and confidence interval on the correlation matrix plot.
library(corrplot)
testRes.B = cor.mtest(variables.B, conf.level = 0.95)

testRes.F = cor.mtest(variables.F, conf.level = 0.95)

# plotting the correlation heatmap
ggplot(data = corr.df.B, aes(x=Var1, y=Var2, 
                                   fill=value)) + 
geom_tile() +
geom_text(aes(Var2, Var1, label = value), 
          color = "black", size = 4)

# Save plot
ggsave("corplot.Allvars.png", corplot.B, 
  width = 16,
  height = 12,
  units = "cm",
  dpi = 600)

```
## Correlation heatmap using microViz package functions
### Bacteria
```{r}

# Recode water surce
ps1.16S <- ps1.16S %>%
  ps_mutate(water.source = if_else(water.source == "Municipal", true = 1, false = 0))

#To get all core taxa not only shared taxa (intersect)
BAC.finaltax.Core <- subset_taxa(ps1.16S, Genus %in% c(Bactaxa.95.all))

# Agglomarate by genus
BAC.finaltax.Core <- tax_agg(BAC.finaltax.Core, "Genus")

# make simple correlation heatmap with all numeric-like variables
cor.heatmap.B <- cor_heatmap(
  data = BAC.finaltax.Core, taxa = NA, cor = "pearson",
  vars = c("plant.age", "pH", "EC", "water.temp", "humidity"),
  grid_lwd = 3,
    tax_anno = taxAnnotation(
      Prev. = anno_tax_prev(ylim = 0:1),
      CLR = anno_tax_box(trans = "clr", zero_replace = "halfmin")
    ), taxa_side = "left")

#width = grid::unit(60, "mm"), height = grid::unit(12, "cm")

# Can't save the plot

```

### Fungi
```{r}
#To get all core taxa not only shared taxa (intersect)
FUN.finaltax.Core <- subset_taxa(ps1.ITS, Genus %in% c(Funtaxa.95.all))


# Agglomarate by genus
FUN.finaltax.Core <- tax_agg(FUN.finaltax.Core, "Genus")

# make simple correlation heatmap with all numeric-like variables
cor.heatmap.F <- cor_heatmap(
  data = FUN.finaltax.Core, taxa = NA, cor = "pearson",
  vars = c("plant.age", "pH", "EC", "water.temp", "humidity"),
  grid_lwd = 3,
    tax_anno = taxAnnotation(
      Prev. = anno_tax_prev(ylim = 0:1),
      CLR = anno_tax_box(trans = "clr", zero_replace = "halfmin")
    ), taxa_side = "left")

#width = grid::unit(60, "mm"), height = grid::unit(12, "cm")

```


